pkg/controller/replication_controller.go

```diff
@@ -72,6 +74,20 @@ func (r RealPodControl) createReplica(namespace string, controller api.Replicati
 		desiredAnnotations[k] = v
 	}
 
+	createdByRef, err := api.GetReference(&controller)
+	if err != nil {
+		util.HandleError(fmt.Errorf("unable to get controller reference: %v", err))
+		return
+	}
+
+	createdByRefJson, err := json.Marshal(createdByRef)
```

Comment:
@simon3z @bgrant0607 This serializes our internal, unversioned object. These annotations will not be readable as soon as that changes!

This should use some codec's Encode function.